src/auth/auth.service.ts

```diff
@@ -24,4 +27,50 @@ export class AuthService {
     });
     return newUser
   }
+
+  
+  async signout(refreshToken: string) {
+    let check = await this.userRepo.findOne({ where: { refreshToken: refreshToken }})
+
+    if (!check) {
+      throw new BadRequestException(
+        ZaLaResponse.BadRequest(
+          'Invalid Refresh Token',
+          'Get the correct refresh token and try again'
+        )
+      );
+    }
+
+    await this.userRepo.update(
+      { refreshToken: refreshToken}, {refreshToken: null}
+    );
+    return ZaLaResponse.Ok('', 'Logged out successfully', 201)
+  }
+
+  
+  async changepassword(id:string, dto: ChangePasswordDto) {
+    const user = await this.userRepo.findOne({ where: { id: id}})
+    const currentPasswordHash = user.password;
+
+    const oldPasswordHash = await this.jwtHelperService.hashPassword(
+      dto.oldPassword,
+      user.password.split(':')[0]
+    );
+
+    if(currentPasswordHash !== oldPasswordHash){
+      throw new BadRequestException(
+        ZaLaResponse.BadRequest(
+          `Access Denied!`,
+          `The password provided doesn't match`,
+          `401`
+        ),
+      );
+    }
+
+    const newPasswordHash = await this.jwtHelperService.hashPassword(
+      dto.newPassword,
+      user.password.split(':')[0]
+    )
+    return await this.userRepo.update(id, {password:newPasswordHash})
+  }
```

Comment:
when a user wants to change password there must be verification that the user is logged in.. i think change password should not precede the sign-in flow.. the refresh token and access token needs to be checked for validity before a user can be allowed to change the password

Reply:
@TijanAyo @SunmadeBabafemi is right, we might even include sending an otp to ensure maximum security

Reply:
But we need a decorator for that @SunmadeBabafemi that will secure all sensitive endpoints like this one. so we can ignore this for now, or what do you think?

Reply:
yes it can be done sir